ooniprobe/View/OONIRun/OoniRunViewController.m

```diff
@@ -49,27 +50,35 @@ -(void)handleUrlScheme{
      NSLog(@"dict: %@", dict);
     
     NSString *action;
-    if ([[url host] isEqualToString:@"run.ooni.io"])
+    NSUserDefaults *usrInfo = [[NSUserDefaults alloc] initWithSuiteName:@"group.org.openobservatory.ooniprobe"];
+
+    if ([[url host] isEqualToString:@"run.ooni.io"]) {
         action = [[url path] substringFromIndex:1];
```

Comment:
I don't think this diff is correct. Sadly, the original code lacks documentation about what it is doing here.

Though, AFAICT, the first case in the `if` (i.e., the one checking for the host being `run.ooni.io`) is the case of links obtained from `run.ooni.io`. The third case (i.e., the one where we check whether the host is equal to `nettest` _could_ be an ancient format used by OONI Run, but I am not sure).

What I find wrong is that, before the change you introduced, there was a check for the minimum version performed over the parameters. Now, this check _only_ applies to the third case (host equal to `nettest`) and not to the first case (host equal to `run.ooni.io`), which however is _the common case_ for modern OONI Run links.

I would suggest to keep the original algorithm and adding the case of `usrInfo` contains `incomingURL` perhaps before and entirely as a separate code path.

I would also suggest to try to document a bit better what we are doing here. My impression is that we do not fully understand what this class is doing, so it seems a good time to figure out where is the OONI Run documentation and ensure that this code references the documentation.

Reply:
When OONI Run is launched from the browser, the actual URL which reaches the app is `ooni://nettest?tn=web_connectivity&ta=...`. On processing, the actual hostname is `nettest`.